### DISCUSSION TITLE
Customize metering options

### DISCUSSION THREAD

**Discussion by RikshaDriver** [gopro/labs#619] [Feature Requests]
This is likely going to be a multi-feature request as there are a few points that need to be discussed for potential development.

Right now there appear to be two options when it comes to exposure metering:

- Multi-Metering (default)
- Spot Metering

Given the nature of use when it comes to GoPro cameras, there can be wildly varying exposure levels in a single frame. I've done a fair amount of testing using gray charts and have observed that with an overall evenly lit scene, proper exposures relative to mid-gray can vary wildly between -1 2/3 EV and -1/2 EV if using the standard Multi-Metering. This goes some way towards explaining why there are complaints that the image appears darker when using a proper technical transform or LUT.

With GP-Log / LogB400 -2 EV shots, this could mean that the whole scene may potentially be up to 4 stops under-exposed... which may go some way towards explaining complaints regarding noise with manual grading.

**Request # 1: Allow multi-metering with some bias... i.e. Shadow weighted bias or Highlight weighted bias - this would allow for some weighting preference based on the dynamic nature of a scene.**

We next have Spot Metering, but it's not entirely clear if this is actual spot metering or center weighted metering.

**Request # 2: Depending on what is already implemented, add either proper spot metering or add center weighted metering with adjustable box size**

It doesn't appear there's a way to set "spot metering" as the default. Enabling "Spot Meter" doesn't appear to survive a reboot either.

**Request # 3:  Allow custom metering mode persistence**

**Comment by dnewman-gpsw** [comment]
GP-Log with a -2EV default is being addressed in the next main firmware release, a few days away (I think.)  

I agree with your findings, GoPro generally biases toward highlight protections, and technical Rec709 LUTs can look dark.  Then the camera uses GTM/LTM to boost anything back up in Native and Vibrant modes.  You can see the camera underexposing when you set Native/Vibrant and then use oMTONE=3 to disable the tone mapping.  

#1 Interesting idea, might be beyond Labs to pull that off.  I will investigate.

#2 Spot meter exists, it is a little hard to make persist, but you can has spot metering to a boot command script to enabled it with each camera start.  From https://gopro.github.io/labs/control/custom/
![image](https://github.com/gopro/labs/assets/10856084/a05a2b15-131b-4ce9-96d0-47eb79efacea)

#3 depends on #1

**Reply by RikshaDriver** [reply]
Thanks David,

Would love to see # 1 implemented as that would fix a lot of auto-exposure related gripes. I exclusively use Protune Flat or GP-Log so the GTM/LTM wouldn't really apply in my case.

Regarding # 2, the current spot metering is about -2/3 EV under-exposed, which is why I'm wondering if it is center weighted with a defined box size.

**Reply by dnewman-gpsw** [reply]
You can do a global EV bias (EVBS), so you could compensate for you preferred exposure.   Try oMEVBS=0.667

**Reply by RikshaDriver** [reply]
Couple of findings...

I have to use oMEVBS="0.6667" with the quotes to get it working.

Interestingly whilst the offset works with LOGB=400, if one uses LOGB=113, a different value of global offset is required.

Strange?

**Reply by lyricsdude** [reply]
Request #2 is a very interesting idea. I thought about it myself recently. I tried vlogging with the spot meter on, because my face was often blown out or underexposed, but if my face is a bit out of the center it exposes on the background right away.

It would be nice a centered box for exposure, and then adjustable in size 1,2,3 or something.

**Comment by dnewman-gpsw** [comment]
The example oMEVBS="0.6667", that is not correct, as using quotes will not interpret the data as a number, EVBS will be ignored.

**Reply by RikshaDriver** [reply]
Thanks, will do some more testing.

For some reason I've been getting wildly different results with Mid-Gray charts on the HERO11 between shots compared to HERO12.

HERO12 seems consistently accurate at oMEVBS=1 with spot metering across GP-Log and Protune Flat.

Will provide some more data after a bit more testing.
